src/cart_cell.cc

```diff
@@ -148,33 +157,113 @@ Real3 CarTCell::CalculateDisplacement(const InteractionForce* force,
 
   Real3 translation_velocity_on_point_mass{0, 0, 0};
 
-  // We check for every neighbor object if they touch us, i.e. push us
-  // away and agreagate the velocities
+  //--------------------------------------------
+  // CAR-T self motility (in case of migration)
+  //--------------------------------------------
+  Real3 current_position = GetPosition();
+  auto* ctxt = sim->GetExecutionContext();
+  auto* rng = sim->GetRandom();
```

Comment:
Can you spell out the type here? The reader should be able to deduce the type reading the right hand side. Here this is not the case.